apps/ui/src/Components/Search/SearchBar.tsx

```diff
@@ -0,0 +1,61 @@
+import TextField from '@mui/material/TextField';
+import Autocomplete from '@mui/material/Autocomplete';
+import { Box, IconButton } from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+import { Value } from 'maplibre-gl';
+import { useRef } from 'react';
+
+export interface SearchProps {
+  options: string[];
+  handleChangeSearchValue: (value: string | undefined) => void;
+}
+
+const SearchBar: React.FC<SearchProps> = ({
+  options,
+  handleChangeSearchValue,
+}) => {
+  const textRef: React.Ref<any> = useRef();
+
+  const handleChange = (_event: React.SyntheticEvent, value: Value) => {
+    const searchValue = value ? value?.toString() : undefined;
+    handleChangeSearchValue(searchValue);
+  };
+
+  const handleClick = (
```

Comment:
Why do you need both `handleChange` and `handleClick`?

Reply:
Because it's two different ways to search, one handles the filter options that open from the autocomplete and the other performs a submit for the search

Basically, the autocomplete comes with submit only by enter, so I added that it would be possible to press a button - more useful on the phone